src/Frame.js

```diff
@@ -0,0 +1,67 @@
+function Frame(ball) {
+  this.ball = ball || new Ball();
+  this.firstRoll = null;
+  this.secondRoll = null;
+  this.score = null;
+  this.scoreNeedsUpdating = false;
+  this.TOTPINS = 10
+}
+
+Frame.prototype.rollBall = function () {
+  var rollValue = this.ball.roll(this._pinsLeft());
+  this.score += rollValue;
+  ((this.firstRoll !== null) ? (this.secondRoll = rollValue) : (this.firstRoll = rollValue))
```

Comment:
Line is too long.<br>Expected an assignment or function call and instead saw an expression.